routing/routing.go

```diff
@@ -136,10 +136,13 @@ func (r *router) route(routes routeSlice, input net.HardwareAddr, src, dst net.I
 		if rt.Src != nil && !rt.Src.Contains(src) {
 			continue
 		}
-		if rt.Dst != nil && !rt.Dst.Contains(dst) {
-			continue
+		if rt.Dst != nil {
+			if !rt.Dst.Contains(dst) {
+				continue
+			}
+			return int(rt.OutputIface), rt.Gateway, rt.PrefSrc, nil
 		}
-		return int(rt.OutputIface), rt.Gateway, rt.PrefSrc, nil
```

Comment:
If you do this, then the default route doesn't match any longer (default route will have both `rt.Src` and `rt.Dst` equal to `nil`) - that just works by coincidence in your case.